### ISSUE TITLE
Autosnippet timeout

### ISSUE THREAD

**Issue by petobens** [L3MON4D3/LuaSnip#634]
Hi! I have the following autosnippet to insert single quotes:
```lua
        s({ trig = 'sq', dscr = 'Single quotes' }, {
            t("'"),
            i(1),
            t("'"),
            i(0),
        }),
```
The problem with this is that if I type something like `square` I obviously get `''`. Thus my requests is if we can we add some sort of timeout for autosnippets such that if I only type `s` and wait, say 1 second, then I can type `q` and the autosnippet won't be triggered?
Thanks in advance.

**Comment by L3MON4D3** [comment]
Phuuh, yeah that would be possible, but I'd really recommend to pick more uncommon triggers, especially for autosnippets. You could also try using conditions to further limit the positions where that snippet will expand

**Comment by petobens** [comment]
I tend to you use uncomon triggers but my muscle memory is hardwired with this one. Anyway if you don't plan to implement a timeout then feel free to close (I was only asking this because I'm migrating from ultisnips and it does have such behaviour for anonymous snippets).

**Comment by L3MON4D3** [comment]
I think we can add this as a setting, can't see any downsides at least (very small, self contained, if ultisnips has that behaviour it's probably a good idea :D )

**Comment by Dioprz** [comment]
@petobens Hi man, sorry, did you find any alternative for this? Something like de-expanding the autosnippet or bypassing the expansion somehow? I also have some autosnippets that I would like to avoid manually.

**Comment by petobens** [comment]
Nope. I believe @L3MON4D3 will eventually implement a setting for it as per his last comment.

**Comment by L3MON4D3** [comment]
Any day now :P
(It will probably still take a while, I'm not all that interested in implementing it, srry :/
If one of you guys wants to look into it, I'll definitely help)

**Comment by Dioprz** [comment]
Nothing to apologize for, on the contrary, thank you very much for still having it on the agenda.

I am actually intrigued to see how complicated it might be to implement the feature, but I will only be able to review it until November/December. If the matter is still open until then, I'll be in touch to get the gist of it.

Thanks again!
